aws/retry/middleware_test.go

```diff
@@ -131,10 +132,11 @@ func TestAttemptMiddleware(t *testing.T) {
 	}()
 
 	cases := map[string]struct {
-		Request testRequest
-		Next    func(retries *[]retryMetadata) middleware.FinalizeHandler
-		Expect  []retryMetadata
-		Err     error
+		Request        testRequest
+		Next           func(retries *[]retryMetadata) middleware.FinalizeHandler
+		Expect         []retryMetadata
+		Err            error
+		ExpectMetadata requestAttempts
 	}{
```

Comment:
It would also be helpful to have a test case that asserts the `Response` value is valid and for an attempt metadata.

Reply:
Also didn't see it but should, if not already, should have a test case for nested `middleware.Metadata`